uk_election_ids/metadata_tools.py

```diff
@@ -25,6 +26,20 @@ def __init__(
 
     DATA = {"defaults": {}}
 
+    def _escape_id_part(self, id_part: str) -> str:
+        r"""
+        Allow use of our slightly modified patterns in the ID requirements json file
+        by escaping operator literals and appending additional operators to patterns where needed.
+
+        In our use case, '.' should represent a string literal (not just a match any)
+        and '*' should represent any number of any character (not just a repeater)
+
+        e.g. 'parl.*.by' becomes the slightly more esoteric 'parl\..*\.by'
+        """
+        id_part = id_part.replace(".", r"\.")  # full-stops converted to string literals
+        id_part = id_part.replace("*", ".*")  # asterisk uses "any number of any character" behaviour
```

Comment:
Do you think this would make more sense as `[^\.]+` so we just wildcard the ID part with the `*` in?

Reply:
I think this comment is about an older version of the regex (or at least, that's what GitHub is showing me)
The most recent iteration 49d2b2d, updated to support the 2018/19 trials is:
```python
id_part = id_part.replace("-", r"\-") 
id_part = id_part.replace(".*.", r"\.(.*\.)?")
```
The above ensures at least one '.' is included in the wildcard but only checks for a second dot if the wildcard has content. This is to avoid matcher fails where a constituency isn't provided for any of the trial authorities but still keep it simple in the json file.

Reply:
Your solution doesn't match some IDs, e.g. `parl.*.by` does not match `parl.reading.tilehurst.by.2023-02-14` but it's an easy fix if you change the repeater from `?` to `*`. Happy to change to use your capture group with that little tweak :)

Reply:
Ah, that's reasonable, sorry I missed it!

Reply:
> `parl.reading.tilehurst.by.2023-02-14`

Of course this ID isn't valid, but your point stands for `local.reading.tilehurst.by.2023-02-14`. Let's stick with your initial solution :sunglasses: